xLights/models/XmlSerializer.h

```diff
@@ -280,22 +512,180 @@ struct XmlSerializingVisitor : BaseObjectVisitor
+        submodels->AddAttribute(XmlNodeKeys::SubBufferStyleAttribute, "foobarbaz");
+        submodels->AddAttribute(XmlNodeKeys::Line0Attribute, "foobarbaz");
+        AddAliases(submodels, aliases);
+        node->AddChild(submodels);
+    }
+
+    void AddGroups(wxXmlNode* node, std::vector<std::string> groups, const std::string name) {
+        for (const auto& g : groups) {
+            wxXmlNode* groups = new wxXmlNode(wxXML_ELEMENT_NODE, XmlNodeKeys::GroupNodeName);
+            groups->AddAttribute(XmlNodeKeys::mgNameAttribute, g);
+            groups->AddAttribute(XmlNodeKeys::mgModelsAttribute, name);
+            groups->AddAttribute(XmlNodeKeys::mgLayoutGroupAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgSelectedAttribute, "foo");
+            groups->AddAttribute(XmlNodeKeys::mgLayoutAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgGridSizeAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgTagColourAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgCentreMinxAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgCentreMinyAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgCentreMaxxAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgCentreMaxyAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgCentrexAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgCentreyAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgCentreDefinedAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgDefaultCameraAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgxCentreOffsetAttribute, "foobarbaz");
+            groups->AddAttribute(XmlNodeKeys::mgyCentreOffsetAttribute, "foobarbaz");
+            // AddAliases(groups,aliases);
+            node->AddChild(groups);
+        }
+    }
+
+    void AddAliases(wxXmlNode* node, const std::list<std::string> aliases) {
+        wxXmlNode* aliashdr = new wxXmlNode(wxXML_ELEMENT_NODE, "Aliases");
+
+        for (const auto& a : aliases) {
+            wxXmlNode* alias = new wxXmlNode(wxXML_ELEMENT_NODE, "alias");
+            alias->AddAttribute("name", a);
+            aliashdr->AddChild(alias);
+        }
+        node->AddChild(aliashdr);
+    }
+
+    void Visit(const ArchesModel& arch) override {
+        wxXmlNode* archNode = new wxXmlNode(wxXML_ELEMENT_NODE, XmlNodeKeys::ModelNodeName);
+
+        const std::vector<Model*>& submodels = arch.GetSubModels();
+        const std::list<std::string>& aliases = arch.GetAliases();
+        const std::string name = arch.GetName();
+
+        const Model* m = dynamic_cast<const Model*>(&arch);
+        const ModelManager& mgr = m->GetModelManager();
+        std::vector<std::string> groups = mgr.GetGroupsContainingModel(m);
+        std::vector<Model*> modelGroups = mgr.GetModelGroups(m);
```

Comment:
@AzGilrock so I can get to the list of groups per your suggestion before, so I can create the group with the proper name and model (Arch in this case), but when I create the group itself, I need to pull in more info from the group itself, center, buffer, preview, tag color, etc... but I can't get it since that's a ModelGroup object vs teh ArchesModel object that I'm in. Dynamic casting doesn't work (at least couldn't figure out a way to do it).

Same for SubModel, but for that I created some more functions to return what I need, but still same issue... can't get from a type ArchesModel to SubModel or ModelGroups.

Any ideas/suggestions?